Beta v8.14.0

MichaIng opened the pull request:
### Beta v8.14.0
_(2022-02-04)_

#### New images
- Orange Pi 5 | Support for Orange Pi 5 SBC has been added to DietPi with hardware ID 80.
- Odroid N2/HC4 | New images have been generated with support for the default Hardkernel SPI bootloader petitboot. Thanks to this, boot from USB devices is now possible, and on Odroid HC4 it is not necessary anymore to erase the SPI flash or flash mainline U-Boot to it. Many thanks to @hstarcm and @Power-onoff for bringing up this topic and help testing adjustments to the boot script: https://github.com/MichaIng/DietPi/issues/5634
- ROCK 5B | New images with ext4 boot filesystem have been generated, to solve possible issues and limitations of the previously used FAT filesystem. Many thanks to @isarrider for reporting this issue: https://github.com/MichaIng/DietPi/issues/6122

#### Enhancements
- NanoPi R2S/NEO3 | Video related kernel modules are now blacklisted on these headless SBCs, saving some bits of RAM and load at boot.
- DietPi-Globals | G_CHECK_NET: When hitting "Retry" from the error handler, the IPv6 connection check is now skipped as well when IPv6 is disabled via dietpi-config network settings or manually. The same is true for changes to the CONFIG_CHECK_CONNECTION/DOMAIN dietpi.txt settings, which can now also be changed right from the error handler. Many thanks to @derSebastian for reporting a related issue: https://github.com/MichaIng/DietPi/issues/5667
- DietPi-Config | It is now possible to change the dietpi.txt network connection test settings CONFIG_CHECK_CONNECTION_IP, CONFIG_CHECK_CONNECTION_IPV6 and CONFIG_CHECK_DNS_DOMAIN via dietpi-config Network Options: Misc menu. Those are used when checking Internet access via IPv4 and IPv6 and DNS resolver functionality.
- DietPi-Software | Gogs: Our ARMv6 build for Raspberry Pi 1 and Zero (1) models has been updated to latest version 0.12.10. The update can be applied via reinstall: dietpi-software reinstall 49

#### Bug fixes
- ROCK Pi S | Resolved an issue where the common serial console on UART0 did not show any output and login prompt. Many thanks to @Marsu31 for reporting this issue: https://github.com/MichaIng/DietPi/issues/5972
- ROCK Pi 4 | Resolved an issue where onboard WiFi on ROCK Pi 4C Plus did not work, and potentially other variants than ROCK Pi 4B. Many thanks to `@Marco` for reporting this issue: https://dietpi.com/forum/t/rock4c-no-wifi-hardware-found/15654
- DietPi-Drive_Manager | Resolved an issue where KVM VirtIO virtual disk volumes were not handled correctly. Many thanks to @bennettp123 for reporting this issue: https://github.com/MichaIng/DietPi/issues/6128
- DietPi-Software | Docker: Resolved a v8.13 regression where Docker dependants (Portainer, Roon Extension Manager) failed to install as the Docker socket was not activated. Many thanks to @omavoss for reporting this issue: https://github.com/MichaIng/DietPi/issues/6065
- DietPi-Software | ownCloud/Nextcloud: Resolved an issue where a global software password with dashes lead to a failing install. Many thanks to @maxmoonshark for reporting this issue: https://github.com/MichaIng/DietPi/issues/4199
- DietPi-Software | myMPD: Resolved an issue where intended initial configs were not applied correctly and web UI changes did not survive a service restart.
- DietPi-Software | X.Org X Server: Resolved an issue where the install failed on ARM VMs with KMS support, where the xserver-xorg-video-vmware package was tried to be installed. Many thanks to @mews-se for reporting this issue: https://github.com/MichaIng/DietPi/issues/6085
- DietPi-Software | Amiberry: Resolved an issue where using keyboard LEDs as power/activity indicators did not work, and stopping the service could hang. You can apply the fix by reinstalling Amiberry via "dietpi-software reinstall 108". Many thanks to @DJP75 for reporting this issue: https://github.com/MichaIng/DietPi/issues/5536

MichaIng committed:
Release v8.13 (#6053)

MichaIng committed:
Init v8.14

Signed-off-by: MichaIng <micha@dietpi.com>

MichaIng committed:
v8.14

- DietPi-Software | Docker: Resolved a v8.13 regression where Docker dependants (Portainer, Roon Extension Manager) failed to install as the Docker socket was not activated. Many thanks to @omavoss for reporting this issue: https://github.com/MichaIng/DietPi/issues/6065

MichaIng committed:
Live patch 0 v8.13 (#6066)

- Live patch 0 | Fix Portainer and Roon Extension Manager installs: https://github.com/MichaIng/DietPi/issues/6065

Signed-off-by: MichaIng <micha@dietpi.com>

MichaIng committed:
v8.14

- DietPi-Software | TasmoAdmin: Update fallback URL

MichaIng committed:
v8.14

- DietPi-Software | Komga: Update fallback URL

MichaIng committed:
v8.14

- DietPi-Software | Gitea: Update fallback URL

MichaIng committed:
v8.14

- DietPi-Software | Komga: Update fallback URL

MichaIng committed:
v8.14

- DietPi-Installer | Replace all mentions of old name "DietPi-PREP"

MichaIng committed:
v8.14

- DietPi-Installer | Add some cleanup for Radxa images

MichaIng committed:
v8.14

- DietPi-Installer | Syntax

MichaIng committed:
v8.14

- DietPi-Software | Move to PHP 8.2 on Bookworm, which is not available with all required modules while PHP 8.1 packages are removed from the repo

MichaIng committed:
v8.14

- DietPi-Software | Nextcloud: Disable for Bookworm on ARMv6/7 since 32-bit is not supported from NC26 on but PHP 8.2 is not supported until NC26.

MichaIng committed:
Live patch 1 v8.13 (#6075)

- Live patch 1 | Fix PHP install on Bookworm by moving to PHP 8.2

MichaIng committed:
v8.14

- DietPi-Services | Handle PHP 8.2 now on Bookworm

MichaIng committed:
v8.14

- DietPi-Installer | On native PC Bookworm images, install zstd for better initramfs compression, and where we know for sure that the kernel supports it
- DietPi-Installer | Harden initramfs-tools install decision: Install it whenever we are not sure that it's not required, based on actually installed kernel instead of on hardware model only

MichaIng committed:
v8.14

- DietPi-Software | Komga: Update fallback URL

MichaIng committed:
v8.14

- DietPi-Software | Minor code format

MichaIng committed:
v8.14

- DietPi-Software | Download_Install: Add support for xz and tgz archives. "tar.xz" is currently not supported. A differenciation between "tar.xz" (contains a directory or multiple files, unpacked with tar command) and "xz" (a single compressed file, decompressed with xz command) can be implemented when needed. Same for gz vs tar.gz and others, where currently only the tarballs are supported.
- DietPi-Software | Gitea: Download xz-compressed file and decompress locally instead of downloading the uncompressed binary

MichaIng committed:
v8.14

- DietPi-Software | Gitea: Update fallback URL

MichaIng committed:
v8.14

- DietPi-Software | ownCloud/Nextcloud: Resolved an issue where a global software password with dashes lead to a failing install. Many thanks to @maxmoonshark for reporting this issue: https://github.com/MichaIng/DietPi/issues/4199

MichaIng committed:
v8.14

- DietPi-Software | Nextcloud: Assure that Nextcloud 25 is downloaded on Bullseye, since Nextcloud 26 won't support PHP 7.4 anymore: https://github.com/nextcloud/server/pull/34997

MichaIng committed:
v8.14

- DietPi-Software | ownCloud/Nextcloud: Make occ/ncc commands aliases instead of shell functions. Those are not used in any other script than dietpi-software anymore, where we already define the shortcut functions. That way it is possible to add bash-completion: https://github.com/nextcloud/server/pull/35451

MichaIng committed:
v8.14

- DietPi-Software | Nextcloud: Patch Nextcloud on Bookworm to support PHP 8.2 and re-enable ARMv6/7 support on Bookworm accordingly, as this allows running Nextcloud 25 on Bookworm.

MichaIng committed:
v8.14

- DietPi-Software | Syntax

MichaIng committed:
v8.14

- DietPi-Software | Add some missing software docs links

MichaIng committed:
v8.14

- CI | Gogs: Add build workflow

MichaIng committed:
v8.13

- CI | Backport Gogs build workflow to master, required for workflow_dispatch trigger

MichaIng committed:
v8.14

- CI | Gogs: APT commands need to run as root user

MichaIng committed:
v8.14

- CI | Gogs: ARMv6 builds need to be done in our container

MichaIng committed:
v8.14

- DietPi-Software | Gogs: Use new ARMv6 archive from our server and install only binary: Everything else is unnecessary init script templates and readme

MichaIng committed:
v8.14

- DietPi-FirstBoot | Assure that service finishes before gettys are started, otherwise it can break autologin in containers where no network is setup
- CI | Gogs: Add temporary workaround for autologin issue by applying above change to the container image

MichaIng committed:
v8.14

- META | Satisfy shellcheck

MichaIng committed:
v8.14

- DietPi-FirstBoot | getty-pre.target is a passive unit and needs to be pulled in

MichaIng committed:
v8.14

- CI | Gogs: Pull in getty-pre.target to order DietPi-FirstBoot before gettys

MichaIng committed:
v8.14

- DietPI-Software | WireGuard: Fix support test and allow in containers, where we have no good change to find out whether the host kernel supports it or not

MichaIng committed:
Merge branch 'master' into dev

MichaIng committed:
v8.14

- META | Remove live patches merged from master

MichaIng committed:
v8.14

- CI | Gogs: Fix path to Gogs binary

MichaIng committed:
v8.14

- CI | Gogs: Fix curl syntax

MichaIng committed:
v8.14

- DietPi-Software | myMPD: Resolved an issue where intended initial configs were not applied correctly and web UI changes did not survive a service restart.

MichaIng committed:
v8.14

- DietPi-Software | Gogs: zip and 7z archives do preserve execute permissions

MichaIng committed:
v8.14

- CI | Gogs: Minor tweaks

MichaIng committed:
v8.14

- CI | Gogs: Switch to matrix build

MichaIng committed:
v8.14

- CI | Gogs: Syntax
- DietPi-Software | Gogs: Use per-distro builds for ARMv6

MichaIng committed:
v8.14

- CI | Gogs: Syntax

MichaIng committed:
v8.14

- CI | Gogs: Fix repo refs

MichaIng committed:
v8.14

- CI | Gogs: Syntax

MichaIng committed:
v8.14

- CI | Gogs: Variable fix

MichaIng committed:
v8.14

- CI | Gogs: Syntax

MichaIng committed:
v8.14

- CI | Syntax

MichaIng committed:
v8.14

- CI | Gogs: Fix quotation

MichaIng committed:
v8.14

- DietPi-Software | X.Org X Server: Resolved an issue where the install failed on ARM VMs with KMS support, where the xserver-xorg-video-vmware package was tried to be installed. Many thanks to @mews-se for reporting this issue: https://github.com/MichaIng/DietPi/issues/6085

MichaIng committed:
v8.14

- DietPi-Software | Readarr: Update fallback URL

MichaIng committed:
v8.14

- DietPi-Software | Gogs: Minor

MichaIng committed:
v8.14

- CHANGELOG | Gogs: Our ARMv6 build for Raspberry Pi 1 and Zero (1) models has been updated to latest version 0.12.10. The update can be applied via reinstall: dietpi-software reinstall 49

MichaIng committed:
v8.14

- DietPi-Software | Minor

MichaIng committed:
v8.14

- DietPi-Installer | ROCK Pi S: Use device tree overlay prefix for current edge kernel package. On "current" kernel, no overlay with this prefix exists yet, but the others do not work anyway, so with this we are future prove.
- DietPi-Installer ROCK Pi S: Fix serial console device, which is ttyS0 on vendor and mainline kernel, it seems to have been ttyS2 only on older mainline kernel images: https://github.com/MichaIng/DietPi/issues/5972. Also remove console=tty1 from kernel cmdline, as this device is headless.
- DietPi-Software | ROCK Pi S: ROCK Pi S is headless as well, hence disable the getty on TTY1
- DietPi-Set_hardware | ROCK Pi S: All UARTs (ttyS0 - ttyS2) on this SBC support a baud rate of 1.500.000: https://wiki.radxa.com/RockpiS/Debian#UART

MichaIng committed:
v8.14

- DietPi-Installer | Usr zstd-compressed initramfs on images with rockchip64 kernel build from Armbian, where we know it's supported, for better compression and faster decompression

MichaIng committed:
v8.14

- META | Satisfy shellcheck

MichaIng committed:
v8.14

- DietPi-Software | SSH servers: Applying SSH password login config implies a restart if the service is active, but in case of Dropbear it is not really running due to NO_START=1 default. And since the sysvinit-systemd wrapper still shows it as active, also the start at the end of the install won't really start it. So assure services are unmasked, enabled and active to really apply all changes by the restart. It is then not required to do that on end of install.

MichaIng committed:
v8.14

- DietPi-Software | Do not allow cleanup jobs to fail. If they do, something is wrong with the filesystem which should be recognised and abort the install, to be repeated once solved.

MichaIng committed:
v8.14

- DietPi-Installer | When checking for legacy kernel, consider Linux 6-9 major versions as well

MichaIng committed:
v8.14

- DietPi-Software | Gitea: Update fallback URL
- DietPi-Build | Add support for NanoPi M3/T3/Fire3

MichaIng committed:
v8.14

- DietPi-Build | Little fix

MichaIng committed:
v8.14

- DietPi-Installer | Download NanoPi M3/T3/Fire3 boot.cmd until it got merged into master
- DIetPI-Installer | Do not follow redirects when accessing raw.githubusercontent.com

MichaIng committed:
v8.14

- DietPi-Installer | Our NanoPi M3/T3/Fire3 bootloader supports ext4 only

MichaIng committed:
v8.14

- DietPi-Installer | Our NanoPi M3/T3/Fire3 bootloader supports MMC devices only

MichaIng committed:
v8.14

- DietPi-Software | PHP: Do not install and offer to remove info pages which provide too much information to be install without admins explicit consent

MichaIng committed:
v8.14

- DietPi-Software | vaultwarden: Bump web vault to v2023.1.1: https://github.com/dani-garcia/bw_web_builds/releases/tag/v2023.1.1

MichaIng committed:
v8.14

- WiFi | From Debian Bookworm on, the crda package is not available anymore. Since Linux 4.15 it is also obsolete, as the kernel is able to load this data directly. It requires however the data stored in /lib/firmware, which is the case from Bullseye (or Buster backports) on. For these cases, we do not not pre-install the crda package anymore, and offer to have it removed on next DietPi update. Our tools are adjusted to check for the existence of a key crda file to know whether to apply the country code to its config file and show the related menu option or not. The only other case, where the country code plays a role, is the WiFi hotspot, to provide it via hostapd. If the WiFi hotspot is installed but crda is now, the country code can still be selected, but is installed to the hostapd config only and not to crda or the interface directly (iw reg set).

MichaIng committed:
v8.14

- DietPi-Globals | G_AGUP: From Bullseye on, use -eany flag to about as well on transient errors e.g. due to failing DNS resolution
- DietPi-Globals | Minor coding and wording

MichaIng committed:
v8.14

- DietPi-Set_hardware | Do not enable boot output on USB getget serial device, which requires loading a kernel module first and hence cannot be used for boot outputs
- DietPi-Installer | On NanoPi M3/T3/Fire3, enable USB gadget serial device and enable login prompt for it. This is very handy when powering the system via USB from PC.

MichaIng committed:
v8.14

- DietPi-Software | myMPD: From next version on, there will new config key to disable plain HTTP: https://github.com/jcorporation/myMPD/commit/17dc906

MichaIng committed:
v8.14

- DietPi-Installer | Fix accident

MichaIng committed:
v8.14

- DietPi-Installer | Sparky SBC: Do not install initramfs-tools, install new kernel modules cleanly and pre-configure Lighttpd as default browser, since Apache cannot start anymore since Bullseye, due to too old kernel version: https://dietpi.com/forum/t/allogui-not-working-apache-service-doesnt-start/15708

MichaIng committed:
v8.14

- DietPi-Build | No need to disable serial consoles on Sparky SBC, as this is done by dietpi-installer already

MichaIng committed:
v8.14

- DietPi-Build | Update required log dirs in Allo GUI image

danielpinto8zz6 committed:
v8.14 (#6099)

- Image | Orange Pi 5 support added

MichaIng committed:
v8.14

- DietPi-Build | On Sparky SBC, apply workaround to fix Roon Bridge start, as nAmbientCapabilities is not supported by the old kernel

MichaIng committed:
v8.14

- DietPi-Software | InfluxDB: Update repo key URL: https://github.com/influxdata/influxdb/pull/24061
- DietPi-Software | Jellyfin: Use bookworm repo suite on Bookworm systems, which is now available
- DietPi-Pre-patches | Update InfluxDB APT repo key
- DietPI-Pre-patches | Update Jellyfin repo suite on Bookworm
- DietPi-Pre-patches | Some repordering and coding enhancements

MichaIng committed:
v8.14

- DietPi-Globals | G_AG*: Pass input arguments through in all cases, to allow testing/passing apt-get options
- DietPi-Globals | G_AGUG: Allow upgrades of packages if those requiere the installation of new packages (without the need to remove any packave). This is especially relevant for kernel upgrades on x86_64 systems where new kernel versions are hold back as the meta package linux-image-amd64 depends on a versioned kernel package like linux-image-6.1.0-1-amd64, which is hence newly installed. As kernel upgrades can contain security relevant patches, and on x86_64 there are max patch version upgrades released (save to apply without breaking changes), this is the better default behaviour and aligns x86_64 DietPi systems with ARM systems, where kernel package names are not versioned.

Joulinar committed:
v8.14 (#6105)

- DietPi-Software | Koel: Install fixed v5.1.14 on PHP 7.x distro versions as detecting latest v5 via api.github.com is not possible anymore

MichaIng committed:
v8.14

- DietPi-Software | Komga: Update fallback URL

MichaIng committed:
v8.14

- DietPi-Software | Amiberry: Resolved an issue where using keyboard LEDs as power/activity indicators did not work, and stopping the service could hang. You can apply the fix by reinstalling Amiberry via "dietpi-software reinstall 108". Many thanks to @DJP75 for reporting this issue: https://github.com/MichaIng/DietPi/issues/5536

MichaIng committed:
v8.14

- CI | Minor build script alignment, and merge G_AGI with G_AGDUG, which is now possible

MichaIng committed:
v8.14

- CI | Unmerge G_AGDUG and G_AGI until next release, since our build containers run on master branch

MichaIng committed:
v8.14

- DietPi-Software | Amiberry: Tiny

Joulinar reviewed (commented):
Are you sure that this is correct to add `=`? At least following is suggested by the error message

```
maintenance:install [--database DATABASE] [--database-name DATABASE-NAME] [--database-host DATABASE-HOST]         
[--database-port DATABASE-PORT] [--database-user DATABASE-USER] [--database-pass [DATABASE-PASS]]                 
[--database-table-space [DATABASE-TABLE-SPACE]] [--admin-user ADMIN-USER] [--admin-pass ADMIN-PASS] [--admin-email
[ADMIN-EMAIL]] [--data-dir DATA-DIR]              
```

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -4612,7 +4627,7 @@ The install script will now exit. After applying one of the the above, rerun die
 
 				}
-				G_EXEC ncc maintenance:install --no-interaction --database 'mysql' --database-name 'nextcloud' --database-user 'tmp_root' --database-pass "${nc_password//?/X}" --admin-user "$username" --admin-pass "${GLOBAL_PW//?/X}" --data-dir "$datadir"
+				G_EXEC ncc maintenance:install --no-interaction --database='mysql' --database-name='nextcloud' --database-user='tmp_root' --database-pass="${nc_password//?/X}" --admin-user="$username" --admin-pass="${GLOBAL_PW//?/X}" --data-dir="$datadir"
```
Are you sure that this is correct to add `=`? At least following is suggested by the error message.

```
maintenance:install [--database DATABASE] [--database-name DATABASE-NAME] [--database-host DATABASE-HOST]         
[--database-port DATABASE-PORT] [--database-user DATABASE-USER] [--database-pass [DATABASE-PASS]]                 
[--database-table-space [DATABASE-TABLE-SPACE]] [--admin-user ADMIN-USER] [--admin-pass ADMIN-PASS] [--admin-email
[ADMIN-EMAIL]] [--data-dir DATA-DIR]
```

**EDIT**
Unsing an alternative command without `=` is working well

```
[FAILED] DietPi-Software | Nextcloud ncc install
 - Command: ncc maintenance:install --no-interaction --database=mysql --database-name=nextcloud --database-user=tmp_root --database-pass=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX --admin-user=admin --admin-pass=XXXXXXXX --data-dir=/mnt/dietpi_userdata/nextcloud_data
[ INFO ] DietPi-Software | Executing alternative command: ncc maintenance:install --no-interaction --database mysql --database-name nextcloud --database-user tmp_root --database-pass XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX --admin-user admin --admin-pass XXXXXXXX --data-dir /mnt/dietpi_userdata/nextcloud_data
Nextcloud was successfully installed
```

MichaIng replied:
Can you retry with latest commit? Should now be fixed.

Joulinar replied:
Yes way better now 👍

MichaIng committed:
v8.14

- DietPi-Software | Nextcloud: Fix internal password replacement for occ install, which is done to avoid having it printed to console

MichaIng committed:
v8.14

- DietPi-Globals | G_AGUP: Do not pass input arguments through for now, since this causes issues when using "-f [file]"/"-v [variable]" to store number of upgradable packages

MichaIng committed:
v8.14

- DietPi-Software | rTorrent: Update ruTorrent fallback URL

MichaIng committed:
v8.14

- DietPi-AutoStart | Resoled an issue where option 17 (custom script foreground with autologin) did not apply since the service was incorrectly removed instead of created.

MichaIng committed:
v8.14

- DietPi-AutoStart | Reverts commit 507b68e820e4c6ee0527c4544fb4530b80602958 mostly, as the option is implemented via DietPi-Login :)

MichaIng committed:
v8.14

- DietPi-Login | Tiny

MichaIng committed:
v8.14

- DietPi-Software | Transmission: Add global software password directly as hash to the config file. Many thanks to @hanscees for pointing us to the right references to make this possible.

MichaIng committed:
v8.14

- DietPi-Software | Transmission: The salt needs to be appended to the hashed password as well

MichaIng committed:
v8.14

- DietPi-Software | Unbound: Add code comment about the necessity to install dns-root-data since Bookworm

MichaIng committed:
v8.14

- DietPi-Software | Nextcloud: Apply PHP 8.2 patch as well on reinstalls to cover distro upgrades from Bullseye to Bookworm: https://github.com/MichaIng/DietPi/pull/6103#issuecomment-1407767853

MichaIng committed:
v8.14

- DietPi-Installer | Do not enforce ROCK Pi 4B device tree in our ROCK Pi 4 image, to enable onboard WiFi support on ROCK Pi 4C+ and potentially fix other issues on other variants

MichaIng committed:
v8.14

- NanoPi R2S/NEO3 | Blacklist video related kernel modules

MichaIng committed:
v8.14

- DietPi-Patches | Purge CRDA on Bookworm without asking, since it cannot work. However, switch to upstream version of regulatory database since the Debian version is accepted only by Debian kernels: https://bugs.debian.org/969897

MichaIng committed:
v8.14

- DietPi-Patches | Do not reapply /etc/modprobe.d/dietpi-headless.conf since updating the initramfs takes so long...

MichaIng committed:
v8.14

- DietPi-Installer | Enforce upstream regolatory database if WiFi is used, to be most compatible with all kernels
- DietPi-Software | Navidrome: Update fallback URL

MichaIng committed:
v8.14

- DietPi-Patches | Move software update info to respective patches

MichaIng committed:
v8.14

- Network | Mark wireless-regdb as manually installed in any case, so it doesn't get accidentally autoremoved whenever CRDA is uninstalled for another reason
- Network | Use upstream regulatory database in any case from Bullseye on, to be compatible with any non-Debian kernel. The reason for the Debian-only variant is to be found out: https://bugs.debian.org/969897
- Network | When applying a country code, do not restart the network and check back anymore. If CRDA is even still supported by the kernel (otherwise manually applying the country code cannot work), and is not overridden by the kernel's "new" (since Linux 4.15 on Bullseye an up) builtin feature, there are various reasons it can fail or require a WiFi reconnection. This reconnetion is however done when selecting "Apply" for IP settings already.
- DietPi-Set_hardware | wificountrycode: Apply dietpi.txt setting if not input is given. Adjust function to gracefully handle missing CRDA and missing hostapd, just apply to what is there

MichaIng committed:
v8.14 (#6103)

- META | Add script for distro upgrade to Bookworm

MichaIng committed:
v8.14

- DietPi-Set_hardware | Re-add execute bit to this script

MichaIng committed:
v8.14

- DietPi-Config | When restarting the network, only restart DHCP server and hostapd instead of all services and only prompt for purging WiFi packages, if any is installed
- DietPi-Config | WiFi Hotspot: Restore IPv6 iptabls NAT rules as well
- DietPi-Config | WiFi country code: Update shown country code when changed, and obtain it from hostapd or CRDA config, if present, only else via iw.

MichaIng committed:
v8.14

- Network | Switch from wireless-tools to iw commands, aside of for SSID scanning where iw scan can (NanoPi Fire3 with modern USB WiFi adapter) hang forever without producing any output...

MichaIng committed:
v8.14

- DietPi-Installer | Install and use zstd on Bullseye x86_64 systems as well for initramfs compression

MichaIng committed:
v8.14

- DietPi-Survey_report | Add CPU time results to report page, even if no CPU temps are available for a certain upload or even a whole hardware model ID. This can be the case if a kernel does not provide the related sysfs entry or our globals function is not able to find it. The CPU speeds would be still nice to know/share.
- DietPi-Survey_report | Enhance bench results processing a bit
- DietPi-Survey_report | Fix DietPi version sorting

MichaIng committed:
v8.14

- DietPi-Software | NoMachine: Bump to v8.3.1

MichaIng committed:
v8.14

- DietPi-Installer | Do not install Debian WiFi/BT firmware if it was selected to be not required

MichaIng committed:
v8.14

- DietPi-Globals | G_AGUP: Do not show arguments in command output

MichaIng committed:
v8.14

- DietPi-Set_software | Do not install CRDA if kernel version cannot be detected. It is very unlikely to be required then as it is a container/chroot or similar and Linux <4.15 is rare anyway, luckily

MichaIng committed:
v8.14

- DietPi-Software | ownCloud: Adjust password replacement to new key=value way of calling occ install, done to avoid printing password to console

MichaIng committed:
v8.14

- DietPi-Software | myMPD: Remove obsolete config key: https://github.com/jcorporation/myMPD/releases/tag/v10.2.1

MichaIng committed:
v8.14

- DietPi-Installer | Add hint why fdtfile needs to be defined for ROCK Pi S

hello-jason committed:
v8.14 (#6120)

- DietPi-Software | WordPress: Fix letter casing
- README | WordPress: Add GitHub URL to software list

Co-authored-by: MichaIng <micha@dietpi.com>

MichaIng committed:
v8.14

- DietPi-Globals | G_CHECK_NET: When hitting "Retry" from the error handler, the IPv6 connection check is now skipped as well when IPv6 is disabled via dietpi-config network settings or manually. The same is true for changes to the CONFIG_CHECK_CONNECTION/DOMAIN dietpi.txt settings. Many thanks to @derSebastian for reporting a related issue: https://github.com/MichaIng/DietPi/issues/5667

MichaIng committed:
v8.14

- DietPi-Config | Rework menu handling to allow direct exiting from any sub menu with respective CLI arguments. Align an in case enhance the cancel/back/exit button texts according to whether the script will exit, return to a parent menu or a direct value selection is cancelled, to make clear that nothing will change. The code has been restructured as well for this change.

MichaIng committed:
v8.14

- DietPi-Globals | Do not assign/call an array to/within a script (thanks shellcheck)

danielpinto8zz6 committed:
v8.14 (#6124)

- Orange Pi 5 | Keep overlay prefix in sync with Armbian

MichaIng committed:
v8.14

- DietPi-Config | Remove firmware upgrade option. It is obsolete since "apt upgrade" as well as now G_AGUG do kernel package upgrades, even if new packages need to be installed for that. "apt full-upgrade"/"apt-get dist-upgrade"/G_AGDUG is required only if packages need to be removed for an upgrade.
- DietPi-Config | Start adding menus to change all network connection test parameters

MichaIng committed:
v8.14

- DietPi-Build | ROCK 5B: Use ext4 filesystem for boot partition
- DietPi-Installer | Enable zstd initramfs compression on RK3588 and meson64 kernels as well

MichaIng committed:
v8.14

- DietPi-Installer | ROCK 5B: Allow setup with Armbian kernel as well with ext4 boot partition

MichaIng committed:
v8.14

- dietpi-bookworm-upgrade | Fix info about how to proceed (or abort) after upgrade and before migrations. Show only first 4 lines of /etc/os-release and add a newline before halt message. Prevent DietPi-Backup prompt on reinstalls as a backup was alread offered before the upgrades which would then be overwritten (if no multiple backup iterations were chosen).

MichaIng committed:
v8.14

- DietPi-Build | Minor changes to boot script

MichaIng committed:
v8.14

- DietPi-Survey_report | Fix DietPi version sorting

MichaIng committed:
v8.14

- DietPi-Software | Docker: The APT repo provides a Bookworm suite now

MichaIng committed:
v8.14

- DietPi-Software | Tiny

MichaIng committed:
v8.14

- DietPi-Config | Remove $2 EXITONBACK input parameter. DietPi-Config will now exit from given $2 TARGETMENUID automatically. This is the most expected behaviour and $2=1 was used in all cases where we used $1 in our scripts.
- DietPi-Config | Do not show exit confirmation dialogue. The additional tab+hit is annoying (IMO) and from whereever you exit DietPi-Config, you can just call it again if it was an accident.
- DietPi-Config | Remove reboot dialogue when locales or proxy settings are changed. A reboot isn't required, but changes are effective for any new bash session, i.e. logout and login is sufficient.
- DietPi-Config | Finish implementation of menus to adjust network connection test parameters
- DietPi-Config | Minor coding enhancements

MichaIng committed:
v8.14

- DietPi-Software | Nextcloud: Unblock Nextcloud 26 on 32-bit ARM systems, since the decision to drop support for them has been reverted: https://help.nextcloud.com/t/update-on-32-bit-support-decision-to-block-the-update-on-nextcloud-26-reverted/153345

MichaIng committed:
v8.14

- DietPi-Build | Tiny enhancement to U-Boot script

MichaIng committed:
v8.14

- DietPi-Globals | G_CHECK_NET: Network connection test parameters can now be changed right from the error handler
- CHANGELOG | DietPi-Config: It is now possible to change the dietpi.txt network connection test settings CONFIG_CHECK_CONNECTION_IP, CONFIG_CHECK_CONNECTION_IPV6 and CONFIG_CHECK_DNS_DOMAIN via dietpi-config Network Options: Misc menu. Those are used when checking Internet access via IPv4 and IPv6 and DNS resolver functionality.

MichaIng committed:
v8.14

- DietPi-Config | Fix exiting network connection check dialogues

MichaIng committed:
v8.14

- DietPi-Config | Fix default item for network connection check menus

MichaIng committed:
v8.14

- DietPi-Installer | Do not try to source user fixup script, if it failed to load to memory space
- DietPi-Installer | Odroid N2 and HC4: Enable support for USB boot via petitboot. This requires to set the ${fdtfile} variable, possible based von ${variant}, forcing petitboot to leave root=UUID=... kernel command-line arguments untouched (by default it turns them into /dev/... entries, but with swapped device paths for SD card and eMMC), using another command to import environment files via ini sections, and avoiding "test -e" conditions.

MichaIng committed:
v8.14

- DietPi-Installer | Odroid N2/HC4: Download new boot.cmd from dev branch until DietPi v8.14 has been released, so we can generate master-based images with this feature. Also satisfy shellcheck.

MichaIng committed:
v8.14

- META | Fix satisfying shellcheck

MichaIng committed:
v8.14 (#6129)

- DietPi-Build | Enable F2FS support for GitHub Actions runners
- DietPi-Build | Replace hardcoded filesystem check and resize commands (NanoPi R5S/R6S and Allo GUI image builds) with rootfs depending variables

MichaIng committed:
v8.14

- DietPi-Software | Readarr: Update fallback URL

MichaIng committed:
v8.14

- DietPi-Software | Pi-hole: Add security headers to responses from web interface
- DietPi-Software | Apache: Remove obsolete X-Download-Options header

MichaIng committed:
v8.14

- DietPi-Software | Navidrome: Update fallback URL

MichaIng committed:
v8.14

- DietPi-Drive_Manager | Resolved an issue where KVM VirtIO virtual disk volumes were not handled correctly. Many thanks to @bennettp123 for reporting this issue: https://github.com/MichaIng/DietPi/issues/6128

MichaIng committed:
v8.14

- RC up

MichaIng committed:
v8.14

- DietPi-Software | Prowlarr: Update fallback URL

MichaIng committed:
v8.14

- Odroid N2/HC4 | Add ini section to dietpiEnv.txt for petitboot support

MichaIng committed:
v8.14

- CHANGELOG | ROCK 5B: New images with ext4 boot filesystem have been generated, to solve possible issues and limitations of the previously used FAT filesystem. Many thanks to @isarrider for reporting this issue: https://github.com/MichaIng/DietPi/issues/6122
